helfomat-infrastructure-jpa/src/main/java/de/helfenkannjeder/helfomat/infrastructure/jpa/JpaDomainEventListener.java

```diff
@@ -0,0 +1,34 @@
+package de.helfenkannjeder.helfomat.infrastructure.jpa;
+
+import de.helfenkannjeder.helfomat.core.organisation.event.OrganisationEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Valentin Zickner
+ */
+@Component
+public class JpaDomainEventListener {
+
+    private final static Logger LOG = LoggerFactory.getLogger(JpaDomainEventListener.class);
+
+    private final EventRepository eventRepository;
+
+    public JpaDomainEventListener(EventRepository eventRepository) {
+        this.eventRepository = eventRepository;
+    }
+
+    @EventListener
+    public void listen(OrganisationEvent organisationEvent) {
+        LOG.debug("Received domain event {}", organisationEvent);
+        this.eventRepository.save(
+            new Event(
+                new EventId(),
```

Comment:
Why does the user have to create this id? In my recent jpa experience, I always use the following to automatically generate the ids as UUIDs. 

```
  @Id
    @Column(name = 'event_id')
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID eventId;
```

You could also create an additional constructor without the ID field and generate the ID in the entity to remove this burden from the user of the entity.

Reply:
This doesn't work with having value objects for your ids. However, I think that whoever creates the entities should be allowed to choose the UUID and it should be only rejected if it's invalid. It has the advantage that you can use it as a command pattern and whoever creates it doesn't need to wait for a response to know the actual id of the object.

Reply:
But you could add a second constructor without the field so it is automatically generated. This way users can specify it themselves but don't have to care if they don't care.